src/middle-pgsql.cpp

```diff
@@ -577,13 +577,15 @@ void middle_pgsql_t::start()
     } else {
         // (Re)create tables.
         m_db_connection.exec("SET client_min_messages = WARNING");
-        for (auto &table : m_tables) {
-            fmt::print(stderr, "Setting up table: {}\n", table.name());
-            auto const qual_name = qualified_name(
-                table.m_copy_target->schema, table.m_copy_target->name);
-            m_db_connection.exec(
-                "DROP TABLE IF EXISTS {} CASCADE"_format(qual_name));
-            m_db_connection.exec(table.m_create);
+        for (auto const &table : m_tables) {
+            if (!table.m_create.empty()) {
+                fmt::print(stderr, "Setting up table: {}\n", table.name());
+                auto const qual_name = qualified_name(
+                    table.m_copy_target->schema, table.m_copy_target->name);
+                m_db_connection.exec(
+                    "DROP TABLE IF EXISTS {} CASCADE"_format(qual_name));
```

Comment:
I think we should still try to drop the table here, just in case that it was left over from a previous run without flatnode store.

Reply:
Should we check whether there is data in there before dropping it? Or maybe just warn the user?

Reply:
We've never done this so far.

Reply:
Fixed in a5a2642